ts/features/walletV3/onboarding/components/WalletOnboardingError.tsx

```diff
@@ -1,42 +1,79 @@
 /* eslint-disable functional/immutable-data */
 import * as React from "react";
-import * as O from "fp-ts/lib/Option";
-import { pipe } from "fp-ts/lib/function";
 
-import { OnboardingOutcome } from "../types";
+import { OnboardingOutcome, OnboardingOutcomeEnum } from "../types";
 import I18n from "../../../../i18n";
-import { GenericErrorComponent } from "../../common/components/GenericErrorComponent";
+import { OperationResultScreenContent } from "../../../../components/screens/OperationResultScreenContent";
+import { openWebUrl } from "../../../../utils/url";
+import {
+  ONBOARDING_FAQ_ENABLE_3DS,
+  ONBOARDING_OUTCOME_ERROR_PICTOGRAM
+} from "../utils";
 
 type WalletOnboardingErrorProps = {
   onClose: () => void;
-  outcome?: OnboardingOutcome;
+  outcome: Exclude<OnboardingOutcome, `${OnboardingOutcomeEnum.SUCCESS}`>;
 };
 
-const getOutcomeMessage = (outcome?: OnboardingOutcome): string =>
-  pipe(
-    outcome,
-    O.fromNullable,
-    O.fold(
-      () => I18n.t("wallet.onboarding.generalError"),
-      outcome => `Outcome ricevuto: ${outcome}` // TODO: Replace with the desired message to be defined
-    )
-  );
+type OnboardingErrorEnumKey = Exclude<
+  keyof typeof OnboardingOutcomeEnum,
+  "SUCCESS"
+>;
 
 /**
- * Component used to show an error message when the wallet onboarding fails
- * TODO: Define the desired design of this component, for now it's just a generic error component
- * that shows an error message or the outcome received
+ * Component used to show an error message when the wallet onboarding fails with a specific outcome
  */
 const WalletOnboardingError = ({
   onClose,
   outcome
-}: WalletOnboardingErrorProps) => (
-  <GenericErrorComponent
-    onClose={onClose}
-    pictogram="error"
-    body={getOutcomeMessage(outcome)}
-    title={I18n.t("wallet.onboarding.failure")}
-  />
-);
+}: WalletOnboardingErrorProps) => {
+  const handleOnPressPrimaryAction = () => {
+    onClose();
+  };
+  const handleOnPressSecondaryAction = () => {
+    switch (outcome) {
+      case OnboardingOutcomeEnum.AUTH_ERROR:
+        openWebUrl(ONBOARDING_FAQ_ENABLE_3DS);
+    }
+  };
+
+  const outcomeEnumKey = Object.keys(OnboardingOutcomeEnum)[
+    Object.values(OnboardingOutcomeEnum).indexOf(
+      outcome as OnboardingOutcomeEnum
+    )
+  ] as OnboardingErrorEnumKey;
+
+  const renderSecondaryAction = () => {
+    switch (outcome) {
```

Comment:
why the single case switches?

Reply:
Because there will be more cases later (for example, [chek this figma](https://www.figma.com/file/FSSrvrUrkAaCxsvse1oFFH/Onboarding-metodi?type=design&node-id=1698-13425&mode=design&t=cnBsVAytXGvtC64S-4)) and so I have already set up the switch-case